File: app/src/main/java/com/ekoapp/simplechat/ChannelMembershipAdapter.java
package com.ekoapp.simplechat;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.ekoapp.ekosdk.EkoChannelMembership;
import com.ekoapp.ekosdk.EkoObjects;
import com.ekoapp.ekosdk.adapter.EkoChannelMembershipAdapter;
import com.ekoapp.simplechat.ChannelMembershipAdapter.EkoUserViewHolder;

import butterknife.BindView;

public class ChannelMembershipAdapter extends EkoChannelMembershipAdapter<EkoUserViewHolder> {

    @Override
    public EkoUserViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_channel_membership, parent, false);
        return new EkoUserViewHolder(view);
    }

    @Override
    public void onBindViewHolder(EkoUserViewHolder holder, int position) {
        EkoChannelMembership channelMembership = getItem(position);
        if (EkoObjects.isProxy(channelMembership)) {
            holder.textView.setText("loading...");
        } else {
            holder.textView.setText(new StringBuilder()
                    .append("id: ")
                    .append(channelMembership.getUserId())
                    .append("\ndisplay name: ")
                    .append(channelMembership.getUser() != null ? channelMembership.getUser().getDisplayName() : "null")
                    .append("\nmembership: ")
                    .append(channelMembership.getMembership())
                    .toString());
        }
    }


    static class EkoUserViewHolder extends BaseViewHolder {

        @BindView(R.id.channel_membership_textview)
        TextView textView;

        EkoUserViewHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/ekoapp/sdk/channellist/ChannelMembershipAdapter.kt
package com.ekoapp.sdk.channellist

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.ekoapp.ekosdk.adapter.EkoChannelMembershipAdapter
import com.ekoapp.ekosdk.channel.membership.EkoChannelMembership
import com.ekoapp.sdk.R
import io.reactivex.BackpressureStrategy
import io.reactivex.Flowable
import io.reactivex.subjects.PublishSubject
import kotlinx.android.synthetic.main.item_channel_membership.view.*

class ChannelMembershipAdapter : EkoChannelMembershipAdapter<ChannelMembershipAdapter.EkoUserViewHolder>() {

    private val onLongClickSubject = PublishSubject.create<EkoChannelMembership>()
    private val onClickSubject = PublishSubject.create<EkoChannelMembership>()

    val onLongClickFlowable: Flowable<EkoChannelMembership>
        get() = onLongClickSubject.toFlowable(BackpressureStrategy.BUFFER)

    val onClickFlowable: Flowable<EkoChannelMembership>
        get() = onClickSubject.toFlowable(BackpressureStrategy.BUFFER)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EkoUserViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.item_channel_membership, parent, false)
        return EkoUserViewHolder(view)
    }

    override fun onBindViewHolder(holder: EkoUserViewHolder, position: Int) {
        val channelMembership = getItem(position)
        if (channelMembership == null) {
            holder.itemView.channel_membership_textview.text = "loading..."
        } else {
            holder.itemView.channel_membership_textview.text = StringBuilder()
                    .append(position + 1)
                    .append("\nid: ")
                    .append(channelMembership.getUserId())
                    .append("\ndisplay name: ")
                    .append(channelMembership.getUser()?.getDisplayName() ?: "null")
                    .append("\nmembership: ")
                    .append(channelMembership.getMembership())
                    .append("\nis muted: ")
                    .append(channelMembership.isMuted())
                    .append("\nis banned: ")
                    .append(channelMembership.isBanned())
                    .append("\nrole: ")
                    .append(channelMembership.getRoles())
                    .toString()

            holder.itemView.setOnClickListener {
                onClickSubject.onNext(channelMembership)
            }

            holder.itemView.setOnLongClickListener {
                onLongClickSubject.onNext(channelMembership)
                true
            }
        }
    }

    class EkoUserViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}
